File: component/src/main/java/com/mackwu/component/ComponentApp.java
package com.mackwu.component;

import android.app.Application;

import com.mackwu.component.func.LocalDataManager;
import com.mackwu.component.util.WebViewUtil;


/**
 * @author MackWu
 * @since 2020/12/7 16:50
 */
public class ComponentApp extends Application {

    public static final String TAG = ComponentApp.class.getSimpleName();
    private static ComponentApp instance;

    @Override
    public void onCreate() {
        super.onCreate();
        instance = this;
        LocalDataManager.getInstance().init(this);
    }

    public static ComponentApp getInstance() {
        return instance;
    }

}

File: component/src/main/java/com/mackwu/component/ComponentApp.kt
package com.mackwu.component

import android.app.Application
import com.mackwu.component.other.db.StudentDao

/**
 * ===================================================
 * Created by MackWu on 2019/11/18 14:23
 * <a href="mailto:wumengjiao828@163.com">Contact me</a>
 * <a href="https://github.com/mackwu828">Follow me</a>
 * ===================================================
 */
class ComponentApp : Application() {

    override fun onCreate() {
        super.onCreate()
        app = this

        //
        StudentDao.init(this)
    }

    companion object {
        lateinit var app: ComponentApp
    }
}